### PR TITLE
Publish API docs on Github Pages (close #73)

### LINKED ISSUE #73: Publish API docs on Github Pages
Generate API docs website using Doxygen or something else and publish on Github Pages in a Github action.

### PR DESCRIPTION
**Pull request by matus-tomlein**
Addresses issue #73 and adds API docs using Doxygen with the [doxygen-awesome theme](https://github.com/jothepro/doxygen-awesome-css) to make it look more modern.

I also added the general docs for the tracker into the /docs folder which is then published in the Doxygen docs along with API docs. I like having it all together, similar to how the [Flutter tracker does it](https://pub.dev/documentation/snowplow_tracker/latest/).

The docs are published on release using Github actions. Here is a test version of the deployed docs: https://snowplow.github.io/snowplow-cpp-tracker/

Btw, the build is failing on Linux because of a problem with apt-get that was fixed in PR #74.

### CODE REVIEW

**Comment by snowplowcla** [comment]
Thanks for your pull request. Is this your first contribution to a Snowplow open source project? Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).

:memo: Please visit https://docs.snowplowanalytics.com/docs/contributing/contributor-license-agreement/ to learn more and sign.

Once you've signed, please reply here (e.g. I signed it!) and we'll verify. Thanks.

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -18,6 +18,36 @@ jobs:
       - name: Run tests
         run: make unit-tests
 
+  docs:
```
Comment (AlexBenny) [inline .github/workflows/release.yml]: Perhaps, we could put the docs in a separate workflow?
It wouldn't be hooked to the release workflow in that case.
Not a big change in any case.

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -18,6 +18,36 @@ jobs:
...
+      - name: Checkout
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+
+      - name: install Doxygen 1.9.3
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y graphviz libclang-cpp1-9 libclang1-9
+          wget https://www.doxygen.nl/files/doxygen-1.9.3.linux.bin.tar.gz
```
Comment (AlexBenny) [inline .github/workflows/release.yml]: I used [mattnotmitt/doxygen-action@v1 action](https://github.com/mattnotmitt/doxygen-action) but looking at this code it seems we could do the same on mobile as it's very simple.

File: docs/01-setup.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+The Tracker is hosted on Github and versions of the Tracker will need to be included by manually dragging and dropping the source files directly into your codebase.
+
+Everything in both the `src` and `include` folders will need to be included in your application. It is important to keep the same folder structure as references to the included headers have been done like so: `../include/json.hpp`.
+
+If you wish to change this you will need to manually update the `src` header files.
+
+**NOTE**: We have included the raw `sqlite3.h & c` files which will need to be compiled separately before being linked to your main C++ project.
+
+The current version of the Snowplow C++ Tracker is 0.2.0.
```
Comment (AlexBenny) [inline docs/01-setup.md]: I guess this is just a test, but in any case  0.2 is obsolete. I guess we are in 0.4.

IMO I would remove this. The versions in the docs become rapidly obsolete and they are easy to forget.
The docs usually already report the version in the title (or something like), so I would avoid to put the version in other places that we (I) easily forget to update.

**Review by AlexBenny — APPROVED** [review]
Looks great 👍 
The Doxygen style is nice, it something we should put on mobile too.
I like the fact that having the official docs written in markdown lets us to use them directly here. It can be a good reference. I'm in doubt if we should replicate all of them here or just a setup page and migration guide, redirecting to the official docs for the rest. It would be useful to keep all the traffic in the official docs for getting feedback or analyse the traffic.

File: .github/workflows/release.yml (hunk 1 above)
Comment (matus-tomlein) [reply .github/workflows/release.yml]: Good idea, separated the workflow for docs out.

File: docs/01-setup.md (hunk 3 above)
Comment (matus-tomlein) [reply docs/01-setup.md]: Good catch, I forgot that it was mentioned. Removed the line from the docs.

**Comment by matus-tomlein** [comment]
Thanks for the review @AlexBenny! I understand the point about guiding all traffic to the official docs, but seeing the problems that we have with the official docs, I wonder if that would only help us and not the users? I think we can discuss this on a meeting.

**Pull request merged**